### ISSUE TITLE
Gizmo Errors game version 0.218.21

### ISSUE THREAD

**Issue by OrianaVenture** [BruceOfTheBow/BruceComfyMods#26]
Running version 1.10.0 with only bepinex installed results in the following. Installing Jotunn version 2.20.3 hides/fixes the issue. Using version 1.9.2 on valheim 218.21 seems to work fine from quick testing.

```
[Info   : Unity Log] 09/13/2024 22:58:05: Worldgenerator version setup:2

[Info   : Unity Log] 09/13/2024 22:58:07: Checking for installed DLCs

[Info   : Unity Log] 09/13/2024 22:58:07: DLC:beta installed:False

[Error  : Unity Log] KeyNotFoundException: The given key '7e6bbee7a32b746cb9396cd890ce7189' was not present in the dictionary.
Stack trace:
System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) (at <17d9ce77f27a4bd2afb5ba32c9bea976>:0)
SoftReferenceableAssets.AssetBundleLoader.GetLoader (SoftReferenceableAssets.AssetID assetID) (at <c63adb3914604620b2f935c5b2d4f66f>:0)
SoftReferenceableAssets.AssetBundleLoader.Load (SoftReferenceableAssets.AssetID assetID) (at <c63adb3914604620b2f935c5b2d4f66f>:0)
SoftReferenceableAssets.SoftReference`1[T].Load () (at <c63adb3914604620b2f935c5b2d4f66f>:0)
ComfyLib.AssetCache`1[T].GetAsset (System.String guid) (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyLib.AssetUtils.GetShader (System.String guid) (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.Gizmos.SetupMaterial (UnityEngine.Material material) (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.Gizmos.SetupComponentsAndRoots () (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.Gizmos..ctor () (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.Gizmos.CreateGizmos () (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.DefaultRotator..ctor () (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.RotationManager.Initialize () (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.GamePatch.StartPostfix () (at <88edf4fb0e4049fa82485dda834357d3>:0)
(wrapper dynamic-method) Game.DMD<Game::Start>(Game)

[Info   : Unity Log] 09/13/2024 22:58:08: Zonesystem Start 1587
```

And when launching a world results in spam

```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
ComfyGizmo.RotationManager.HideGizmos () (at <88edf4fb0e4049fa82485dda834357d3>:0)
ComfyGizmo.PlayerPatch.UpdatePlacementPostfix (Player __instance, System.Boolean takeInput) (at <88edf4fb0e4049fa82485dda834357d3>:0)
(wrapper dynamic-method) Player.DMD<Player::UpdatePlacement>(Player,bool,single)
Player.Update () (at <1f17a254067e41d0b59746dec98ad607>:0)
```

**Comment by redseiko** [comment]
I'm having a hard time reproducing this, but it's most likely related to creating the Gizmos in the constructors too early and not waiting for AssetBundleLoader.Initialized to be true or something.

Making a change to adjust for this even if I can't reproduce.

**Comment by redseiko** [comment]
Whoops nevermind, was on the wrong version. Now able to reproduce.

**Comment by redseiko** [comment]
The root issue is most likely there is no call to `Runtime.MakeAllAssetsLoadable();` at plugin start.

Jotunn does this in their `Awake()` which is why the issue doesn't appear when it's also loaded.

**Comment by redseiko** [comment]
Fixed with `v1.11.0`.

https://thunderstore.io/c/valheim/p/ComfyMods/Gizmo/
